agent/src/IngestDetailsRepository.php

```diff
@@ -66,6 +71,15 @@ public function get(): PromiseInterface
         return $this->ingestDetails ??= $this->refresh();
     }
 
+    public function markQuotaExceeded(): void
+    {
+        $this->quotaExceeded = true;
+
+        $this->loop->cancelTimer($this->refreshTimer); // @phpstan-ignore argument.type
+
+        $this->scheduleRefreshIn(60 * 15);
+    }
```

Comment:
When marked as over quota, we cancel the existing token refresh timer, which could be up to an hour long, and set a 15 minute refresh.